pkg/kubelet/dockertools/manager.go

```diff
@@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
 					break
 				}
 			}
+			if pod.Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
```

Comment:
Why aren't we using DeletionTimestamp here instead? Or do we need to convert that to a duration?

I expected:
1. User sets terminationGracePeriodSeconds or it gets set to 30s by default
2. On deletion, deletionTimestamp is set, to Now+terminationGracePeriodSeconds by default by controllers
3. Upon noticing deletionTimestamp is set, prestop hook is executed
4. After prestop hook completes, stop is called with time deletionTimestamp-Now

Eventually we want to "right align" notifications, but the above seems like a reasonable starting point.